worker/uniter/resolver/opfactory.go

```diff
@@ -119,9 +119,16 @@ func (s *resolverOpFactory) wrapUpgradeOp(op operation.Operation, charmURL *char
 func (s *resolverOpFactory) wrapHookOp(op operation.Operation, info hook.Info) operation.Operation {
 	switch info.Kind {
 	case hooks.PreSeriesUpgrade:
```

Comment:
The on prepare wrapper and on commit wrappers sync the remote and local state. The actual execution of this hook sets the remote state. Which is why the two wrappers are free to set statuses against remote state without waiting for the `RemoteStateWatcher`.